bundles/framework/divmanazer/component/Chart.js

```diff
@@ -49,58 +59,53 @@ Oskari.clazz.define('Oskari.userinterface.component.Chart', function(sandbox, lo
         return dimensions;
     },
     createGridlines: function () {
-        return d3.axisBottom(this.x).ticks(10);
+        return d3.axisBottom( this.x ).ticks(10);
     },
     initScales: function () {
         this.x = d3.scaleLinear()
-        .domain([ d3.min( this.data, function (d) {
+        .domain([ d3.min( this.data, function ( d ) {
                 return d.value;
-            }), d3.max( this.data, function (d) {
+            }), d3.max( this.data, function ( d ) {
                 return d.value;
             })
         ]);
 
         this.y = d3.scaleBand()
-        .domain(this.data.map(function (d) {
+        .domain( this.data.map ( function ( d ) {
             return d.name;
         }));
     },
-    initSelection: function() {
-            var svg = d3.select(this._template.graph[0]).append("svg")
-                .attr("width", this.dimensions.width() + this.dimensions.margin.left + this.dimensions.margin.right)
-                .attr("height", this.dimensions.height() + this.dimensions.margin.top + this.dimensions.margin.bottom)
-                .append("g")
-                .attr("transform", "translate(" + this.dimensions.margin.left + "," + this.dimensions.margin.top + ")"
-            );
-            this.svg = svg;
+    initSVG: function () {
+        var svg = d3.select( this._template.graph[0] ).append("svg")
+            .attr( "width", this.dimensions.width() + this.dimensions.margin.left + this.dimensions.margin.right )
+            .attr( "height", this.dimensions.height() + this.dimensions.margin.top + this.dimensions.margin.bottom )
+            .append( "g" )
+            .attr( "transform", "translate(" + this.dimensions.margin.left + "," + this.dimensions.margin.top + ")" );
+        this.svg = svg;
         return svg;
     },
+    /**
+     * d3 axis are functions that generate svg-elements based on the scale given
+     * @method initAxis
+     *
+     */
     initAxis: function () {
-        // var stateService = this.service.getStateService();
-        // var classificationOpts = stateService.getClassificationOpts(this.activeIndicator.hash);
-        // var colors = this.service.getColorService().getColorsForClassification(classificationOpts, true);
-        var color = "#DC143C";
-        // Init axes
-        this.yAxis = d3.axisLeft(this.y)
+        this.yAxis = d3.axisLeft( this.y )
         .tickSize(10);
 
-        this.xAxis = d3.axisBottom(this.x)
+        this.xAxis = d3.axisBottom( this.x )
         .tickSize(10);
-
-        this.colorScale = d3.scaleQuantize()
-        .domain([0, this.data.length])
-        .range(color);
     },
     /**
      * initializes the chart skeleton without any specific line or bar options
      *
      * @method initChart
      *
      */
-    initChart: function() {
-        var selections = this.initSelection();
+    initChart: function () {
+        var svg = this.initSVG();
```

Comment:
I would change initSVG() to getSVGTemplate(), remove assignment of this.svg from the initSVG() implementation and assign it in here if it's needed.